Sources/Sentry/SentryScope.m

```diff
@@ -87,17 +92,30 @@ - (instancetype)init
 - (instancetype)initWithScope:(SentryScope *)scope
 {
     if (self = [self init]) {
-        self.extraDictionary = scope.extraDictionary.mutableCopy;
-        self.tagDictionary = scope.tagDictionary.mutableCopy;
+        @synchronized(_extraDictionary) {
+            [_extraDictionary addEntriesFromDictionary:[scope extras]];
```

Comment:
We're locking in a private field to this instance, and accessing a private field of another instance outside of that instances lock